### PR TITLE
Change wire_symbol to use Register and idx

### PR DESCRIPTION
**Pull request by dstrain115**
- Change wire_symbol to use Register and idx since the wire_symbol should never need to use the binst of a Soquet.

Part of #728

### CODE REVIEW

File: qualtran/testing.py [hunk 1]
```diff
@@ -236,9 +235,7 @@ def assert_wire_symbols_match_expected(bloq: Bloq, expected_ws: List[str]):
     ws = []
     regs = bloq.signature
     for i, r in enumerate(regs):
-        # note this will only work if shape = ().
-        # See: https://github.com/quantumlib/Qualtran/issues/608
-        ws.append(bloq.wire_symbol(Soquet(i, r)).text)
+        ws.append(bloq.wire_symbol(r, (i,)).text)
```
Comment (mpharrigan) [inline qualtran/testing.py:238]: I don't think this fixes the linked issue. Can you restore the comment

**Review by mpharrigan — APPROVED** [review]
Thanks for this! I suspect this may have been a little tedious

File: qualtran/testing.py (hunk 1 above)
Comment (dstrain115) [reply qualtran/testing.py:240]: Done.  If you have a good example, I can fix the comment too.

Comment (mpharrigan) [reply qualtran/testing.py:240]: how about `Toffoli()`

**Pull request merged**